cpp/perspective/CMakeLists.txt

```diff
@@ -562,28 +580,45 @@ elseif(PSP_CPP_BUILD OR PSP_PYTHON_BUILD)
 
 		target_compile_definitions(psp PRIVATE PSP_ENABLE_PYTHON=1)
 		target_compile_definitions(binding PRIVATE PSP_ENABLE_PYTHON=1)
-		target_compile_options(binding PRIVATE -Wdeprecated-declarations)
 
-        # Link against libarrow 0.15.1 in pyarrow directory for windows
         if (WIN32)
-			target_link_libraries(psp arrow)
-			target_link_libraries(binding arrow)
+			target_compile_definitions(binding PRIVATE WIN32=1)
+			target_compile_definitions(binding PRIVATE _WIN32=1)
+
 			target_link_libraries(psp ${Boost_FILESYSTEM_LIBRARY})
 			target_link_libraries(binding ${Boost_FILESYSTEM_LIBRARY})
+
+			# .dll not importable
+			set_property(TARGET binding PROPERTY SUFFIX .pyd)
 		else()
-            target_link_libraries(psp ${PYTHON_PYARROW_LIBRARIES})
-            target_link_libraries(binding ${PYTHON_PYARROW_LIBRARIES})
+			target_compile_options(binding PRIVATE -Wdeprecated-declarations)
             set_property(TARGET psp PROPERTY INSTALL_RPATH ${CMAKE_INSTALL_RPATH} ${module_install_rpath}  ${PYTHON_PYARROW_LIBRARY_DIR})
             set_property(TARGET binding PROPERTY INSTALL_RPATH ${CMAKE_INSTALL_RPATH} ${module_install_rpath}  ${PYTHON_PYARROW_LIBRARY_DIR})
-        endif()
+		endif()
+
+		target_link_libraries(psp ${PYTHON_PYARROW_LIBRARIES})
+		target_link_libraries(binding ${PYTHON_PYARROW_LIBRARIES})
 
 		target_link_libraries(psp ${PYTHON_LIBRARIES})
 		target_link_libraries(binding ${PYTHON_LIBRARIES})
-		target_link_libraries(binding psp)
+
+		target_link_libraries(psp tbb)
 		target_link_libraries(binding tbb)
 
+		target_link_libraries(binding psp)
+
 		add_custom_command(TARGET psp POST_BUILD COMMAND ${CMAKE_COMMAND} -E copy $<TARGET_FILE:psp> ${PSP_PYTHON_SRC}/table/)
 		add_custom_command(TARGET binding POST_BUILD COMMAND ${CMAKE_COMMAND} -E copy $<TARGET_FILE:binding> ${PSP_PYTHON_SRC}/table/)
+
+		if(WIN32)
+			# inline arrow dlls
+			file(GLOB ARROW_DLLS "${PYTHON_PYARROW_LIBRARY_DIR}/*.dll")
```

Comment:
@sc1f @texodus  open to suggestions to get around copying all arrow dlls into our directory. In a "production" environment with a custom arrow build you won't have to worry about this, but with the packaged arrow wheel bringing a ton of stuff with it, we need to ensure all the links resolve.

Reply:
I don't understand what the requirement being solved by copying is, nor why a ton of stuff being brought is the instigator, so I'm not really in a position to offer constructive advice here.  Copying is bad of course.